rubix-core/src/main/java/com/qubole/rubix/core/CachedReadRequestChain.java

```diff
@@ -49,11 +49,13 @@
   private DirectBufferPool bufferPool;
   private int directBufferSize;
   private int corruptedFileCount;
+  private int generationNumber;
```

Comment:
There are multiple places where generationNumber is getting initialized with the default value of int ie 0. I suggest we make '0' as the special case of UNKNOWN_GENERATION_NUMBER and assert in `getLocalPath` and `getMetadataFilePath` to not get this gen number. So valid gen numbers will start from 1 and we can easily catch issues of invalid generationNumber being used